File: dotnet/ArqLibrarianClassic/ArcLibrarianClassis.ATests/Startup/ApplicationStartup.cs
﻿using ArcLibrarianClassis.ATests.Library;
using NUnit.Framework;

namespace ArcLibrarianClassis.ATests.Startup
{
    [TestFixture()]
    public class ApplicationStartup
    {
        private LibraryFixture fixture;

        [SetUp]
        public void SetupFixture()
        {
            this.fixture = new LibraryFixture();
        }


        [Test()]
        public void ShouldShowHelloAtTheBegining()
        {            
            fixture.ApplicationStarted();
            fixture.HasSampleBooks();

            fixture.Then();
            fixture.SystemShows("Welcome to the ArqLibrarian");
        }
    }
}

File: java/ArqLibraryClassic/ArqLibraryLayeredATests/src/bnsit/ms/arq/startup/ApplicationStartup.java
package bnsit.ms.arq.startup;

import bnsit.ms.arq.library.LibraryFixture;
import org.junit.Before;
import org.junit.Test;

public class ApplicationStartup {

    private LibraryFixture fixture;

    @Before
    public void setupFixture() {
        this.fixture = new LibraryFixture();
    }

    @Test
    public void shouldShowWelcomeAtTheBeginning() {
        //given

        //when
        fixture.applicationStarted();
        fixture.hasSampleBooks();


        fixture.then();
        fixture.systemShows("Welcome to the ArqLibrarian");
    }
}
